spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -186,22 +185,24 @@ public ScanBuilder newScanBuilder(CaseInsensitiveStringMap options) {
       icebergTable.refresh();
     }
 
-    SparkScanBuilder scanBuilder = new SparkScanBuilder(sparkSession(), icebergTable, options);
-
-    if (requestedSchema != null) {
-      scanBuilder.pruneColumns(requestedSchema);
-    }
-
-    return scanBuilder;
+    return new SparkScanBuilder(sparkSession(), icebergTable, addSnapshotId(options));
```

Comment:
I think this method needs to determine the base schema (whatever was returned by `schema()`) and pass that into the scan builder so that the scan is consistent with analysis that's already happened using just the table instance.

Reply:
`SparkTable#schema` already returns `snapshotSchema()` converted to Spark's `StructType`. Here, as you suggest, I construct the `SparkScanBuilder` with the `snapshotSchema()`. So we are consistent.